src/components/button/button.tsx

```diff
@@ -77,109 +89,75 @@ export interface ButtonGroupProps extends HTMLAttributes<HTMLDivElement> {
 
 export const Button: FC<ButtonProps> = ({
   children,
-  isPrimary,
-  isSecondary,
-  isTertiary,
-  isDestructive,
-  isBusy = false,
-  isToggle = false,
-  isPressed,
+  variant,
   size = 'medium',
-  iconOnly,
   iconLeft,
   iconRight,
   onClick,
   ariaLabel,
-  isDisabled = false,
+  disabled = false,
   href,
   target,
   rel,
   className,
   ...rest
 }) => {
-  const variant =
-    (isPrimary && elButtonPrimary) ||
-    (isSecondary && elButtonSecondary) ||
-    (isTertiary && elButtonTertiary) ||
-    (isDestructive && elButtonDestructive) ||
-    '' // Default to empty - secondary Button style applied to default
-
   const sizeClass = cx(
-    iconOnly && elButtonIconOnly,
     size === 'small' && elButtonSizeSmall,
     size === 'medium' && elButtonSizeMedium,
     size === 'large' && elButtonSizeLarge,
   )
 
   const miscellaneousClass = cx(
-    isBusy && elButtonBusy,
-    isDisabled && elButtonDisabled, // UI for disabled Anchor element
-    !children && isBusy && elButtonIconOnly, // if no children(label) and is busy then add el-button-icon-only class
+    disabled && elButtonDisabled, // UI for disabled Anchor element // Need to remove this.
+    !children && elButtonIconOnly, // if no children(label) then add el-button-icon-only class
   )
 
-  // Determine the icon font size based on the button size
-  const iconFontSize = fontSizeMap[size]
+  const combinedClassName = cx(className, sizeClass, miscellaneousClass)
 
-  const combinedClassName = cx(className, sizeClass, variant, miscellaneousClass)
-
-  console.log('c', combinedClassName)
-
-  if (href) {
+  if (!isButtonAsButtonElement({ href })) {
     return (
       <ElAnchorButton
-        href={isDisabled || isBusy ? undefined : href}
+        href={href}
+        data-variant={variant}
         className={combinedClassName}
         onClick={(e) => {
-          if (isDisabled) {
+          if (disabled) {
             e.preventDefault() // Prevent default action if disabled
           } else {
-            handleButtonClick(onClick)(e)
+            handleButtonClick(onClick)
```

Comment:
**note:** Instead of using a helper function like `handleButtonClick`, you could just do `onClick?.()`.